### PR TITLE
Feature: disable mutiple patient selection and warning messages for missing files

### PR DESCRIPTION
**Pull request by tdlinh2712**
This PR:
- disable user from selecting multiple patients to open.
- display warning messages for when RTStruct, RTImage or RTPlan files are missing

### CODE REVIEW

**Review by sjswerdloff — COMMENTED** [review]
LGTM

File: src/View/OpenPatientWindow.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def setup_ui(self, open_patient_window_instance):
         self.open_patient_directory_prompt = QLabel()
         self.open_patient_directory_prompt.setObjectName("OpenPatientDirectoryPrompt")
         self.open_patient_directory_prompt.setAlignment(Qt.AlignLeft)
-        self.open_patient_window_instance_vertical_box.addWidget(self.open_patient_directory_prompt);
+        self.open_patient_window_instance_vertical_box.addWidget(self.open_patient_directory_prompt)
 
         # Create a horizontal box to hold the input box for the directory and the choose button
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:43]: Comment length exceeds 72 characters

File: src/View/OpenPatientWindow.py [hunk 2]
```diff
@@ -70,7 +70,8 @@ def setup_ui(self, open_patient_window_instance):
 
         # Create a horizontal box to hold the stop button and direction to the user on where to select the patient
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:71]: Comment length exceeds 72 characters

File: src/View/OpenPatientWindow.py [hunk 3]
```diff
@@ -70,7 +70,8 @@ def setup_ui(self, open_patient_window_instance):
...
         self.open_patient_appear_prompt_and_stop_horizontal_box = QHBoxLayout()
-        self.open_patient_appear_prompt_and_stop_horizontal_box.setObjectName("OpenPatientAppearPromptAndStopHorizontalBox")
+        self.open_patient_appear_prompt_and_stop_horizontal_box.setObjectName(
+            "OpenPatientAppearPromptAndStopHorizontalBox")
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:74]: Line length exceeds 99 characters

File: src/View/OpenPatientWindow.py [hunk 4]
```diff
@@ -91,31 +92,34 @@ def setup_ui(self, open_patient_window_instance):
         self.open_patient_appear_prompt_and_stop_horizontal_box.addWidget(self.open_patient_window_stop_button)
         # Create a widget to hold the layout
         self.open_patient_appear_prompt_and_stop_widget = QWidget()
-        self.open_patient_appear_prompt_and_stop_widget.setLayout(self.open_patient_appear_prompt_and_stop_horizontal_box)
+        self.open_patient_appear_prompt_and_stop_widget.setLayout(
+            self.open_patient_appear_prompt_and_stop_horizontal_box)
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:96]: Line length exceeds 99 characters

File: src/View/OpenPatientWindow.py [hunk 5]
```diff
@@ -91,31 +92,34 @@ def setup_ui(self, open_patient_window_instance):
...
         self.open_patient_window_instance_vertical_box.addWidget(self.open_patient_appear_prompt_and_stop_widget)
 
-
         # Create a tree view list to list out all patients in the directory selected above
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:99]: Comment length exceeds 72 characters

File: src/View/OpenPatientWindow.py [hunk 6]
```diff
@@ -91,31 +92,34 @@ def setup_ui(self, open_patient_window_instance):
...
         self.open_patient_directory_result_label.setObjectName("OpenPatientDirectoryResultLabel")
         self.open_patient_directory_result_label.setAlignment(Qt.AlignLeft)
         self.open_patient_window_instance_vertical_box.addWidget(self.open_patient_directory_result_label)
 
-
         # Create a horizontal box to hold the Cancel and Open button
         self.open_patient_window_patient_open_actions_horizontal_box = QHBoxLayout()
-        self.open_patient_window_patient_open_actions_horizontal_box.setObjectName("OpenPatientWindowPatientOpenActionsHorizontalBox")
+        self.open_patient_window_patient_open_actions_horizontal_box.setObjectName(
+            "OpenPatientWindowPatientOpenActionsHorizontalBox")
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:122]: Line length exceeds 99 characters

File: src/View/OpenPatientWindow.py [hunk 7]
```diff
@@ -137,13 +141,15 @@ def setup_ui(self, open_patient_window_instance):
...
+        self.open_patient_window_confirm_button.setDisabled(True)
         self.open_patient_window_confirm_button.clicked.connect(self.confirm_button_clicked)
         self.open_patient_window_confirm_button.setProperty("QPushButtonClass", "success-button")
         self.open_patient_window_patient_open_actions_horizontal_box.addWidget(self.open_patient_window_confirm_button)
 
         # Create a widget to house all of the actions button for open patient window
         self.open_patient_window_patient_open_actions_widget = QWidget()
-        self.open_patient_window_patient_open_actions_widget.setLayout(self.open_patient_window_patient_open_actions_horizontal_box)
+        self.open_patient_window_patient_open_actions_widget.setLayout(
+            self.open_patient_window_patient_open_actions_horizontal_box)
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:152]: Line length exceeds 99 characters

File: src/View/OpenPatientWindow.py [hunk 8]
```diff
@@ -137,13 +141,15 @@ def setup_ui(self, open_patient_window_instance):
...
         self.open_patient_window_instance_vertical_box.addWidget(self.open_patient_window_patient_open_actions_widget)
 
         # Set the vertical box fourth element, the tree view, to stretch out as far as possible
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:155]: Comment length exceeds 72 characters

File: src/View/OpenPatientWindow.py [hunk 9]
```diff
@@ -164,11 +170,13 @@ def setup_ui(self, open_patient_window_instance):
         open_patient_window_instance.setCentralWidget(self.open_patient_window_instance_central_widget)
 
         # Set the current stylesheet to the instance and connect it back to the caller through slot
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:172]: Comment length exceeds 72 characters

File: src/View/OpenPatientWindow.py [hunk 10]
```diff
@@ -254,6 +265,49 @@ def on_search_complete(self, dicom_structure):
...
             QMessageBox.about(self, "No files found", "Selected directory contains no DICOM files.")
 
+    def tree_item_changed(self, item, _):
+        """
+            Executes when a tree item is checked or unchecked.
+            If a different patient is checked, uncheck the previous patient.
+            Inform user about missing DICOM files.
+        """
+        selected_patient = item
+        # If the item is not top-level, bubble up to see which top-level item this item belongs to
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:275]: Comment length exceeds 72 characters

File: src/View/OpenPatientWindow.py [hunk 11]
```diff
@@ -254,6 +265,49 @@ def on_search_complete(self, dicom_structure):
...
+        if self.open_patient_window_patients_tree.invisibleRootItem().indexOfChild(item) == -1:
+            while self.open_patient_window_patients_tree.invisibleRootItem().indexOfChild(selected_patient) == -1:
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:277]: Line length exceeds 99 characters

File: src/View/OpenPatientWindow.py [hunk 12]
```diff
@@ -262,15 +316,12 @@ def confirm_button_clicked(self):
         for item in self.get_checked_leaves():
             selected_files += item.dicom_object.get_files()
 
-        if len(selected_files) > 0:
-            self.progress_window = ProgressWindow(self, QtCore.Qt.WindowTitleHint | QtCore.Qt.WindowCloseButtonHint)
-            self.progress_window.signal_loaded.connect(self.on_loaded)
-            self.progress_window.signal_error.connect(self.on_loading_error)
+        self.progress_window = ProgressWindow(self, QtCore.Qt.WindowTitleHint | QtCore.Qt.WindowCloseButtonHint)
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:319]: Line length exceeds 99 characters

File: src/View/OpenPatientWindow.py [hunk 13]
```diff
@@ -137,13 +141,15 @@ def setup_ui(self, open_patient_window_instance):
         self.open_patient_window_confirm_button.resize(self.open_patient_window_confirm_button.sizeHint().width(),
                                                     self.open_patient_window_confirm_button.sizeHint().height())
         self.open_patient_window_confirm_button.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
+        self.open_patient_window_confirm_button.setDisabled(True)
         self.open_patient_window_confirm_button.clicked.connect(self.confirm_button_clicked)
         self.open_patient_window_confirm_button.setProperty("QPushButtonClass", "success-button")
         self.open_patient_window_patient_open_actions_horizontal_box.addWidget(self.open_patient_window_confirm_button)
 
         # Create a widget to house all of the actions button for open patient window
```
Comment (Melinda-V) [inline src/View/OpenPatientWindow.py:149]: Comment length exceeds 72 characters

**Pull request merged**